lighthouse-core/report/html/renderer/report-renderer.js

```diff
@@ -35,6 +35,11 @@ class ReportRenderer {
   renderReport(report, container) {
     // If any mutations happen to the report within the renderers, we want the original object untouched
     const clone = /** @type {LH.ReportResult} */ (JSON.parse(JSON.stringify(report)));
+    // Mutate the UIStrings if necessary (while saving originals)
```

Comment:
should the technical debt comment below be moved up here? :)

It does kind of suck to rely on utils as a mutable singleton. We pass at least a `DOM` into all of our report rendering classes. Should we pass in strings? Or pass the lhr to everything, use `lhr.i18n.rendererFormattedStrings.string` everywhere, and here (since we're mutating the lhr anyways) could set `lhr.i18n.rendererFormattedStrings` to the default strings if not set.

Fine to punt to a future PR, but it would be nice to have a plan before we get too deep.

Reply:
> Or pass the lhr to everything, use lhr.i18n.rendererFormattedStrings.string everywhere, and here (since we're mutating the lhr anyways) could set lhr.i18n.rendererFormattedStrings to the default strings if not set.

This sounds good to me, we have other LHR reaching needs like locale in config settings and such that would be nice. I accept your offer for a followup :)

Reply:
add a TODO?